appinsights/src/channel/memory.rs

```diff
@@ -47,9 +59,10 @@ impl InMemoryChannel {
             Self::send_command(&sender, command);
         }
 
-        if let Some(thread) = self.thread.take() {
-            debug!("Shutting down worker");
-            thread.join().unwrap();
+        match std::mem::replace(&mut self.join_handle, ThreadHandle::None) {
+            ThreadHandle::Thread(thread) => thread.join().unwrap(),
+            ThreadHandle::Future(_thread) => {} // TODO: can we block on this?
```

Comment:
This ties into the open question of cancellation in rust futures. The general pattern to follow here is to have a separate channel for signalling shutdown, and within the future being spawned onto the runtime, `select!`ing on both the channel, and the worker future in a loop.

When the shutdown channel receives a notification, we exit the task, which in turn drop the future.